internals/daemon/access.go

```diff
@@ -68,3 +68,18 @@ func (ac UserAccess) CheckAccess(d *Daemon, r *http.Request, user *UserState) Re
 	// An identity explicitly set to "access: untrusted" isn't allowed.
 	return Unauthorized(accessDenied)
 }
+
+// MetricsAccess allows requests over the HTTP from authenticated users.
+type MetricsAccess struct{}
+
+func (ac MetricsAccess) CheckAccess(d *Daemon, r *http.Request, user *UserState) Response {
+	if user == nil {
+		return Unauthorized(accessDenied)
+	}
+	switch user.Access {
+	case state.MetricsAccess, state.AdminAccess:
```

Comment:
Should `ReadAccess` be allowed here too?
```suggestion
	case state.MetricsAccess, state.ReadAccess, state.AdminAccess:
```

Reply:
I thought I left a comment somewhere and didn't resolve this, but I guess it was in the other metrics PR. I will remove `ReadAccess` and keep `AdminAccess` - I think "admin" should have the highest level of access, including "read" metrics.